### PR TITLE
Enable torch.compile and bf16 optimizations in SubspaceEncoder

### PR DESCRIPTION
**Pull request by Adefioye**
Here, I am resolving [issue 5](https://github.com/chrisjmccormick/shared-subspaces/issues/5) by adding `torch.compile` and `bf16` configurations to the config files to run encoders. These results were obtained on RTX 6000 Ada 48gb VRAM.

Here are the results for SST accuracy:

| Model | Params  | Accuracy (Chris) | Accuracy (Koko) |
|-------|---------|------------------|-----------------|
| MLA   | 12.67M  | 84.75            | 82.34           |
| MLA-O | 12.48M  | 84.63            | 83.72           |

__Findings:__
1. After adding the new configurations, there appears to be some speed up in training runs evident by reduction in training time. 
2. The most notable observation is that pre-training runtime for MLA and MLA-O encoders runs are about the same approximately `1h 2mins`. Is this not indicative of a potential benefit for MLA-O? Is it possible that if we increase head size/ number of layers, we might be able to observe more significant reduction in training time?
3. MLM accuracy for MLA and MLA-O are __55.51%__ and __54.61%__ respectively.

### CODE REVIEW

**Comment by chrisjmccormick** [comment]
Thanks @Adefioye, this is looking great.

---

### Affect on Benchmark

I checked out your branch and did a training run of my own on our MLA-o baseline.

torch.compile made a huge difference. The total runtime for training went from 1h 22min to 47min. (Note that Colab now provides 80GB A100s instead of 40GB, so I don't know if the GPU change contributed to this).

With torch.compile and bf16, the training curves (loss, grad_norm, eval_loss) followed the same path as before, but not quite identical. MLM accuracy was 53.55% before, and 53.45% with those changes.

On SST-2, test accuracy went from 84.63% to 84.40%. I did a second fine-tuning run with torch.compile = False and the test accuracy was 84.06%.

bf16 represents a different set of numbers than fp16, so it makes sense that this would change things. I was surprised to see that torch.compile had an effect, though. 

Altogether, I think this supports what you were suggesting--that the SST-2 benchmark is very sensitive, and that we probably can't compare results across different setups (e.g., two different GPUs). 

---

### Scripts

Before we merge your changes, can you move your runpod setup from /runpod_setup into a subfolder, /infra/runpod

That way I can add /infra/lambda at some point as well without cluttering the top level of the repo.

**Review by chrisjmccormick — APPROVED** [review]
Looks good!
